bach/bach/series/series_json.py

```diff
@@ -524,15 +524,22 @@ def _find_in_json_list(self, filtering_slice: Dict[str, str], is_start: bool):
         return Expression.construct(fmt, slicing_mask, self._series_object)
 
     def get_array_item(self, key: int) -> 'TSeriesJson':
-        """ For documentation, see implementation in parent class :class:`JsonAccessor` """
-        if key >= 0:
-            expression = Expression.construct(f'''JSON_QUERY({{}}, '$[{key}]')''', self._series_object)
-            return self._series_object.copy_override(expression=expression)
-        # case key < 0
-        # BigQuery doesn't (yet) natively support this, so we emulate this.
-        array_len = self.get_array_length()
-        expr_offset = Expression.construct(f'OFFSET({{}} {key})', array_len)
-        expression = Expression.construct('JSON_QUERY_ARRAY({})[{}]', self._series_object, expr_offset)
+        """
+        Returns an item from the json array.
+        The key is treated as a 0-based index. If negative this will count from the end of the array (one
+            based). If the index does not exist this will render None/NULL.
+        This assumes the top-level item in the json is an array
+        """
+        if key < 0:
+            # BigQuery doesn't (yet) natively support this, so we emulate this by reversing the array
+            key = -key - 1
+            array_expr = Expression.construct(
+                "'[' || ARRAY_TO_STRING(ARRAY_REVERSE(JSON_QUERY_ARRAY({})), ', ') || ']'",
+                self._series_object
+            )
+        else:
+            array_expr = self._series_object.expression
+        expression = Expression.construct(f'''JSON_QUERY({{}}, '$[{key}]')''', array_expr)
```

Comment:
I think we can simplify the expression even more by just converting the json into array once and stringify the array with `FORMAT` function and using `SAFE_OFFSET` for avoiding raising errors for index out of range.
```suggestion
        array_expr = f'JSON_QUERY_ARRAY({{}})'
        if key < 0:
            # BigQuery doesn't (yet) natively support this, so we emulate this by reversing the array
            key = -key - 1
            array_expr = f'ARRAY_REVERSE({array_expr})'

        access_array_expr = f'{array_expr}[SAFE_OFFSET({key})]'
        expression = Expression.construct(f"FORMAT('%s', {access_array_expr})", self._series_object)
```

Reply:
Aah that's brilliant, I didn't know about `safe_offset`/`safe_ordinal`

Reply:
true, no need for format